### PR TITLE
[circle-mpqsolver] Implement 'set_save_intermediate'

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit implements 'set_save_intermediate' for the AMPQ task.

Its correctness is tested in https://github.com/Samsung/ONE/pull/10655

Draft: https://github.com/Samsung/ONE/pull/10655
Related: https://github.com/Samsung/ONE/issues/10634

ONE-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

File: compiler/circle-mpqsolver/src/MPQSolver.cpp [hunk 1]
```diff
@@ -24,3 +24,15 @@ MPQSolver::MPQSolver(const std::string &input_data_path, float qerror_ratio,
...
+void MPQSolver::set_save_intermediate(bool save, const std::string &save_path)
+{
+  if (save)
+  {
+    _hooks = std::make_unique<core::DumpingHooks>(save_path);
+  }
+  else
+  {
+    _hooks.reset(nullptr);
+  }
```
Comment (jinevening) [inline compiler/circle-mpqsolver/src/MPQSolver.cpp:37]: It seems that function name already has the meaning save = true.

```suggestion
void MPQSolver::set_save_intermediate(const std::string &save_path)
{
  _hooks = std::make_unique<core::DumpingHooks>(save_path);
}
```

**Review by jinevening — APPROVED** [review]
LGTM

**Comment by jinevening** [comment]
@seanshpark Ping.

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**